api/v1/ytsaurus_webhook.go

```diff
@@ -489,25 +532,24 @@ func (r *Ytsaurus) evaluateYtsaurusValidation(old *Ytsaurus) error {
 }
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *Ytsaurus) ValidateCreate() (admission.Warnings, error) {
-	ytsauruslog.Info("validate create", "name", r.Name)
-
-	return nil, r.evaluateYtsaurusValidation(nil)
+func (r *ytsaurusValidator) ValidateCreate(ctx context.Context, _ runtime.Object) (admission.Warnings, error) {
+	ytsauruslog.Info("validate create", "name", r.Ytsaurus.Name)
+	return nil, r.Ytsaurus.evaluateYtsaurusValidation(ctx, nil)
```

Comment:
The reason webhook tests are failing is that before `ValidateCreate` was called on ytsaurus resource
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/webhook/admission/validator.go#L92

but with a custom validator resource is being passed as an argument instead (suppose it is the same object)
https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/webhook/admission/validator_custom.go#L91

So now `r` is empty struct and all validate methods must be fixed to explicitly receive validated object and old object (in update, in create it is nil) (and I suppose it makes sense to move all the methods here to a new ytsaurusValidator struct).